extensions/sql-bindings/src/services/azureFunctionsService.ts

```diff
@@ -219,27 +219,23 @@ export async function createAzureFunction(node?: ITreeNodeInfo): Promise<void> {
 			.withAdditionalProperties(propertyBag)
 			.withConnectionInfo(connectionInfo).send();
 
+		// check for the new function file to be created and dispose of the file system watcher
+		const timeoutForFunctionFile = utils.timeoutPromise(constants.timeoutAzureFunctionFileError);
+		let functionFilePath = await Promise.race([newFunctionFileObject.filePromise, timeoutForFunctionFile]);
+
 		// prompt user for include password for connection string
-		if (isCreateNewProject) {
+		if (isCreateNewProject && functionFilePath) {
```

Comment:
Need to make sure the azure function project is created prior to adding the connection string to the local.settings.json for the folder the user selected (and added to the workspace).